### ISSUE TITLE
Is there a way to use autosuggest with Apache?

### ISSUE THREAD

**Issue by florianbrinkmann** [10up/ElasticPress#1076]
Hi,

I would like to use the autosuggest feature. In the readme there is a part to modify the nginx config, but I use Apache. Is there a way to use the feature with Apache, or does it only work with nginx? I tried to convert the rules to `.htaccess` rules, but without success. Before I continue I would like to know if this should be possible :)

Thanks in advance,
Florian

**Comment by allan23** [comment]
Hi @florianbrinkmann,
This should be doable with Apache. Ideally, you are creating a proxypass to the Elasticsearch endpoint. 
For more information, here is some documentation on setting one up: https://httpd.apache.org/docs/2.4/howto/reverse_proxy.html

Note, your Apache version may be different so be sure to look at the documentation for your version. Also, it is a good idea to deny requests to your endpoint that use POST (only GET requests should go to the Elasticsearch endpoint).

**Comment by florianbrinkmann** [comment]
Thanks @allan23! Currently trying to get it working without having access to the `httpd.conf` (it is a shared hosting), I will write a follow up here if I find a solution :)

**Comment by florianbrinkmann** [comment]
I found a way for it 🎉 I wrote the following into my `.htaccess` file:

```
SetEnvIf Request_URI ^/es-search$/* ADD_ELASTICSEARCH_BASIC_AUTH
RequestHeader set Authorization "Basic XXXXXXXXXXX" env=ADD_ELASTICSEARCH_BASIC_AUTH
RewriteRule ^es-search$ http://example.com/index/post/_search [P]
```

A few notes:
* The solution for passing basic auth data (the first two lines) is from https://serverfault.com/a/710194 – there is also noted how to generate the base64 string from the user and password, that replaces the `XXXXXXXXXXX`.
* I added `/es-search` as the endpoint for autosuggest in the ElasticPress options.
* The Elasticsearch instance runs with Search Box on Heroku, so maybe it works differnet for other hosts.
